From 583589c3574ffd6e0376579316b30d2a2dcf82f8 Mon Sep 17 00:00:00 2001
From: Ferdinando Terada <Ferdinando.GodoyTerada@windriver.com>
Date: Mon, 23 Dec 2024 17:53:09 -0300
Subject: [PATCH] Adjust timeout for coredns readinessProbe

The timeout value for the readinessProbe of CoreDNS was increased.
This adjustment was necessary to avoid issues during stress testing,
ensuring that the component can properly handle high-load situations
and prevent premature failure in readiness checks.
---
 cluster/addons/dns/coredns/coredns.yaml.base   | 1 +
 cluster/addons/dns/coredns/coredns.yaml.in     | 1 +
 cluster/addons/dns/coredns/coredns.yaml.sed    | 1 +
 cmd/kubeadm/app/phases/addons/dns/manifests.go | 1 +
 4 files changed, 4 insertions(+)

diff --git a/cluster/addons/dns/coredns/coredns.yaml.base b/cluster/addons/dns/coredns/coredns.yaml.base
index 3a0fd7adb72..c8289f7c136 100644
--- a/cluster/addons/dns/coredns/coredns.yaml.base
+++ b/cluster/addons/dns/coredns/coredns.yaml.base
@@ -170,6 +170,7 @@ spec:
             path: /ready
             port: 8181
             scheme: HTTP
+          timeoutSeconds: 5
         securityContext:
           allowPrivilegeEscalation: false
           capabilities:
diff --git a/cluster/addons/dns/coredns/coredns.yaml.in b/cluster/addons/dns/coredns/coredns.yaml.in
index 74b59584bc7..974c8337031 100644
--- a/cluster/addons/dns/coredns/coredns.yaml.in
+++ b/cluster/addons/dns/coredns/coredns.yaml.in
@@ -170,6 +170,7 @@ spec:
             path: /ready
             port: 8181
             scheme: HTTP
+          timeoutSeconds: 5
         securityContext:
           allowPrivilegeEscalation: false
           capabilities:
diff --git a/cluster/addons/dns/coredns/coredns.yaml.sed b/cluster/addons/dns/coredns/coredns.yaml.sed
index 61afbecd9da..563a8980e07 100644
--- a/cluster/addons/dns/coredns/coredns.yaml.sed
+++ b/cluster/addons/dns/coredns/coredns.yaml.sed
@@ -170,6 +170,7 @@ spec:
             path: /ready
             port: 8181
             scheme: HTTP
+          timeoutSeconds: 5
         securityContext:
           allowPrivilegeEscalation: false
           capabilities:
diff --git a/cmd/kubeadm/app/phases/addons/dns/manifests.go b/cmd/kubeadm/app/phases/addons/dns/manifests.go
index 2a2212d5d37..c0be57357e4 100644
--- a/cmd/kubeadm/app/phases/addons/dns/manifests.go
+++ b/cmd/kubeadm/app/phases/addons/dns/manifests.go
@@ -135,6 +135,7 @@ spec:
             path: /ready
             port: 8181
             scheme: HTTP
+          timeoutSeconds: 5
         securityContext:
           allowPrivilegeEscalation: false
           capabilities:
-- 
2.34.1

